internal/fuse/snapshots_dirstruct.go

```diff
@@ -302,10 +303,20 @@ func (d *SnapshotsDirStructure) updateSnapshots(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	// sort snapshots ascending by time (default order is descending)
+
+	// Sort snapshots ascending by time (default order is descending).
+	// This needs to be done before hashing.
 	sort.Sort(sort.Reverse(snapshots))
```

Comment:
How are two snapshots with identical timestamps ordered? afaict that order isn't stable?

Reply:
Good point. Fixed to sort by timestamp, then hash.